controllers/cluster/capabilities.go

```diff
@@ -0,0 +1,46 @@
+package cluster
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	v1 "github.com/openshift/api/config/v1"
+	configv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+)
+
+type Capabilities struct {
+	HasEtcdPDBQuorum, HasMachineAPI bool
+}
+
+// HasMachineAPICapability returns true if the cluster has the MachineAPI Enabled
+func HasMachineAPICapability(config *rest.Config) (bool, error) {
+	configV1Client, err := configv1.NewForConfig(config)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to create cluster version client")
+	}
+	cvs, err := configV1Client.ClusterVersions().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil
```

Comment:
IIUC this use case indicates a problem (i.e I expect to find at least one Cluster Version even if machine isn't supported).
So I think we should probably log an error here.

Reply:
In K8s clusters I (actually the k8s-e2e tests) got the error

> failed to get ClusterVersion: the server could not find the requested resource (get clusterversions.config.openshift.io)

and looking into `ConfigV1Client` it seems intended to interact _only_ with `config.openshift.io` [1], so I thought that what is listing here is not all the Cluster Version, but only those from that Group


[1] https://github.com/openshift/client-go/blob/master/config/clientset/versioned/typed/config/v1/config_client.go#L38

Reply:
> However it got me thinking that we were indeed correct in the assumption that this check is only relevant for OCP.

yes, I think you're right here. At least _the way we use to check MachineAPI exists_ only works in OCP (I wonder if there is another way).

In light of this, I'd go for the option number 1, as it keeps `MachineAPI` signature cleaner IMHO

Reply:
moreover, there is no point anymore in checking both `isOnOpenshit` and `hasMachineAPI` [here](https://github.com/clobrano/node-healthcheck-operator/blob/8b772a9f0a0cff2711019facba57e01c01cfee2c/main.go#L190)